Assignments/ip1/ip1.md

```diff
@@ -0,0 +1,328 @@
+   * - GameMove (applies a move to the game)
+   * - LeaveGame (leaves the game)
+   *
+   * If the command ended the game, records the outcome in this._history
+   * If the command is successful (does not throw an error), calls this._emitAreaChanged (necessary
+   * to notify any listeners of a state update, including any change to history)
+   * If the command is unsuccessful (throws an error), the error is propagated to the caller
+   *
+   * @see InteractableCommand
+   *
+   * @param command command to handle
+   * @param player player making the request
+   * @returns response to the command, @see InteractableCommandResponse
+   * @throws InvalidParametersError if the command is not supported or is invalid.
+   * Invalid commands:
+   * - GameMove, StartGame and LeaveGame: if the game is not in progress (GAME_NOT_IN_PROGRESS_MESSAGE) or if the game ID does not match the game in progress (GAME_ID_MISSMATCH_MESSAGE)
+   * - Any command besides JoinGame, GameMove, StartGame and LeaveGame: INVALID_COMMAND_MESSAGE
+   */
+  public handleCommand<CommandType extends InteractableCommand>(
+    command: CommandType,
+    player: Player,
+  ): InteractableCommandReturnType<CommandType>
+{% endhighlight %}
+</details>
+
+Grading for implementation tasks:
+* Handling JoinGame: 2 points
+* Handling GameMove: 2 points
+* Handling LeaveGame: 2 points
+* Handling StartGame: 2 points
+* Handling invalid commands: 2 point
+
+## Submission Instructions
+Submit your assignment to the instance of Autograder.io running at [neu.autograder.io](https://neu.autograder.io).
+Navigate to [neu.autograder.io](https://neu.autograder.io) in your web browser, click the "Sign in" button, and log in with your Northeastern account.
+You should then see the course listed on the neu.autograder.io home page.
+Please contact the instructors immediately if you have difficulty accessing the course on Autograder.io.
+If you haven't been added to the course roster yet, you can access the assignment page at [this direct link](https://neu.autograder.io/web/project/12).
+
+To submit your assignment: run the command `npm run zip` in the top-level directory of the handout. This will produce a file called `covey-town-townService.zip`. Submit that zip file on Autograder.io.
+
+Autograder.io will provide you with feedback on your submission, but note that it will *not* include any marks that will be assigned after we manually grade your submission for code style (these marks will remain hidden until it is graded). It may take several minutes for the grading script to complete.
+
+Autograder.io is configured to only provide feedback on at most 5 submissions per-24-hours per-student (submissions that fail to run or receive a grade of 0 are not counted in that limit). We strongly encourage you to lint and test your submission on your local development machine, and *not* rely on Autograder.io for providing grading feedback - relying on Autograder.io is a very slow feedback loop.
```

Comment:
The "receive a grade of 0" part of not counting submissions may not be possible to have the autograder handle automatically this time around. The reason is that the implementation and test case grading will be split up in order to use Autograder.io's native mutation testing support. This means that any criteria for rejecting a submission have to be checked in the initial setup. 

My suspicion is that a submission receiving a 0 close to the deadline would be most likely because of failed linting or type checking, which would still cause the submission to be rejected and not count towards the limit. Taking the best submission rather than the most recent further mitigates this. Thoughts?

Reply:
There's another layer/detail to this: If the implementation grading rejects the submission when the student's implementation fails against all the instructor tests, then that would prevent the test case grading from running, but I think we want students to be able to hypothetically just submit test cases and get feedback on them.

Reply:
that sounds fine - could you update the text in the assignment to match once the autograder is configured?